src/Data/Functor/Day/Curried.hs

```diff
@@ -45,6 +46,14 @@ instance Functor g => Functor (Curried g h) where
   fmap f (Curried g) = Curried (g . fmap (.f))
   {-# INLINE fmap #-}
 
+instance (Functor g, g ~ h) => Apply (Curried g h) where
+  Curried mf <.> Curried ma = Curried (ma . mf . fmap (.))
+  {-# INLINE (<.>) #-}
+#if MIN_VERSION_semigroupoids(5,2,2)
```

Comment:
I think we should depend on `semigroupoids >= 5.2.2` to avoid having to depend on this bit of CPP.

Reply:
> I'd hate to break a bunch of build plans for this.

`semigroupoids-5.2.2` and later all compile back to GHC 7.0, so they should be quite permissive with respect to build plans. Do you have a particular thing in mind that would break?

Reply:
@ekmett what is your thought on this?

Reply:
Adding the instance matches the Applicative, so there's no objection from me there. No real opinion on how to mitigate build plan breakage. @RyanGlScott's thoughts seem pretty much spot on though. Especially after a 3 year delay. ;)

Reply:
Yeah, I agree.